src/components/Network.js

```diff
@@ -67,6 +72,30 @@ class Network extends React.Component {
     return null;
   }
 
+  calcSyncProgressFromTimestamps(sync_data) {
+    const first_timestamp = this.state.dag.first_timestamp;
+    const latest_timestamp = this.state.dag.latest_timestamp;
+    const delta = latest_timestamp - first_timestamp;
+    const synced_percent = 100 * (sync_data.synced_timestamp - first_timestamp) / delta;
+    let general_percent = (100 * (sync_data.latest_timestamp - first_timestamp) / delta) - synced_percent;
+    if (general_percent > 100) {
+      general_percent = 100;
+    }
+    return { synced_percent, general_percent };
+  }
+
+  calcSyncProgressFromFraction(sync_data) {
+    const { progress = 0 } = sync_data;
+    const synced_percent = Math.min(100, Math.max(0, progress * 100));
+    const general_percent = 100 - synced_percent;
+    return { synced_percent, general_percent };
+  }
+
+  getSyncProgress(sync_data) {
+    return sync_data.synced_timestamp && sync_data.latest_timestamp ?
```

Comment:
We want to always use the new data when it's available, so I would check if we have the new API key `sync` and use the timestamps only if it's no available.

Reply:
You're right. Now it will only use timestamps to build a polyfill of the new sync data structure ([here](https://github.com/HathorNetwork/hathor-explorer/blob/0b8efbf7ba0dafb1e93da0a8b0d888c20ac0f885/src/components/Network.js#L153)).